KeywordSearch/src/org/sleuthkit/autopsy/keywordsearch/ExtractedContentViewer.java

```diff
@@ -104,41 +105,55 @@ public void setNode(final Node node) {
-            if (hits != null) {
-                /*
-                 * if there is a QueryReslt object, in the lookup use that. This
-                 * happens when a user selects a row in an ad-hoc search result
-                 */
-                highlightedHitText = new HighlightedText(content.getId(), hits);
-            } else if (artifact != null
-                    && artifact.getArtifactTypeID() == TSK_ACCOUNT.getTypeID()) {
-                try {
-                    // if the artifact is an account artifact, get an account text .
-                    highlightedHitText = getAccountsText(content, nodeLookup);
-                } catch (TskCoreException ex) {
-                    logger.log(Level.SEVERE, "Failed to create AccountsText for " + content, ex); //NON-NLS
-
+            if (artifact != null) {
+                if (artifact.getArtifactTypeID() == TSK_ACCOUNT.getTypeID()) {
+                    try {
+                        /*
+                         * Generate AccountsText for the account artifact.
+                         */
+                        highlightedHitText = getAccountsText(content, nodeLookup);
+                    } catch (TskCoreException ex) {
+                        LOGGER.log(Level.SEVERE, "Failed to create AccountsText for " + content, ex); //NON-NLS
+                    }
+                } else if (artifact.getArtifactTypeID() == TSK_KEYWORD_HIT.getTypeID()) {
+                    try {
+                        /*
+                         * Generate HighlightedText for the keyword hit artifact.
+                         */
+                        highlightedHitText = new HighlightedText(artifact);
+                    } catch (TskCoreException ex) {
+                        LOGGER.log(Level.SEVERE, "Failed to create HighlightedText for " + artifact, ex); //NON-NLS
+                    }
                 }
-            } else if (artifact != null
-                    && artifact.getArtifactTypeID() == TSK_KEYWORD_HIT.getTypeID()) {
-                try {
-                    //if there is kwh artifact use that to construct the HighlightedText
-                    highlightedHitText = new HighlightedText(artifact);
-                } catch (TskCoreException ex) {
-                    logger.log(Level.SEVERE, "Failed to create HighlightedText for " + artifact, ex); //NON-NLS
+            }
+        } else {
+            /*
+             * Results for ad-hoc search.
+             */
+            QueryContent queryContent = nodeLookup.lookup(QueryContent.class);
+            content = (AbstractFile) queryContent.getContent();
+            
+            if (null != content && solrHasContent(content.getId())) {
+                QueryResults queryResults = queryContent.getResults();
+                if (queryResults != null) {
+                    highlightedHitText = new HighlightedText(queryContent.getSolrObjectId(), queryResults);
```

Comment:
I am not comfortable with the logic in this method, for reasons that are detailed in other comments. Please do the following instead, which assumes that we already know there is indexed text from isSupported, so we can eliminate some Solr queries in this method:
`Query lookup for QueryContent`
`Query lookup for BlackboardArtifact`
`Query lookup for AbstractFile`
`If lookup has QueryContent`
`Construct a HighlightedText using the Solr object id and the query results from the QueryContent` 
`else if lookup has a BlackboardArtifact and it is TSK_KEYWORD_HIT`
`Construct a HighlightedText using the artifact`
`else if lookup has a BlackboardArtifact and it is TSK_ACCOUNT and lookup has AbstractFile`
`Construct a HighlightedText using getAccountsText`
`if lookup has AbstractFile`
`Construct a RawText with file and its id`
`If lookup has a BlackboardArtifact`
`Construct a RawText with getRawArtifactText`

Note that in another story, we will seal up the holes in exception handling so that the user will see error messages when we have TskCoreExceptions, but we will consider that out of scope for this effort.